### ISSUE TITLE
Shift + Click not trading stacks

### ISSUE THREAD

**Issue by AlunKingofWales** [Shopkeepers/Shopkeepers#934]
* **Shopkeepers version:** `2.23.1`
* **Spigot version:** `1.21.3` ( using latest Buildtools 188 )
---

Removed all datapacks and all other plugins, deleted Shopkeepers config.
Clean restarted server.
Joined game as Admin/Op
Used /Shopkeeper command to create a default shop.
Setup a new trade to provide 1 charcoal to the shop, receiving 1 string in return
Interact with the shopkeeper - put a stack of 64 Charcoal into the either of the trade slots - Shift+Click on the String but only trades 1 item from the stack.

Plenty of spare inventory space. Also tried CTRL and ALT clicking but same result, only 1 item traded per click.

Thanks!

**Comment by blablubbabc** [comment]
I am able to reproduce this, getting this error inside the server log:
```
[19:52:19] [Server thread/ERROR]: Could not pass event InventoryClickEvent to Shopkeepers v2.23.1
net.minecraft.ResourceKeyInvalidException: Non [a-z0-9/._-] character in path of location: minecraft:UI_BUTTON_CLICK
	at net.minecraft.resources.MinecraftKey.f(SourceFile:285) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.resources.MinecraftKey.b(SourceFile:60) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.resources.MinecraftKey.a(SourceFile:89) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.resources.MinecraftKey.a(SourceFile:56) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at org.bukkit.craftbukkit.v1_21_R2.entity.CraftPlayer.playSound(CraftPlayer.java:586) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at org.bukkit.craftbukkit.v1_21_R2.entity.CraftPlayer.playSound(CraftPlayer.java:572) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at com.nisovin.shopkeepers.util.bukkit.SoundEffect.play(SoundEffect.java:410) ~[?:?]
	at com.nisovin.shopkeepers.util.bukkit.SoundEffect.play(SoundEffect.java:381) ~[?:?]
	at com.nisovin.shopkeepers.ui.trading.TradingHandler.commonApplyTrade(TradingHandler.java:937) ~[?:?]
	at com.nisovin.shopkeepers.ui.trading.TradingHandler.onInventoryClickLate(TradingHandler.java:586) ~[?:?]
	at com.nisovin.shopkeepers.ui.UIHandler.informOnInventoryClickLate(UIHandler.java:363) ~[?:?]
	at com.nisovin.shopkeepers.ui.UIHandler.informOnInventoryEventLate(UIHandler.java:268) ~[?:?]
	at com.nisovin.shopkeepers.ui.UIListener.onInventoryEventLate(UIListener.java:218) ~[?:?]
	at com.nisovin.shopkeepers.util.bukkit.EventUtils.lambda$eventExecutor$0(EventUtils.java:76) ~[?:?]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-api-1.21.3-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:601) ~[spigot-api-1.21.3-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:588) ~[spigot-api-1.21.3-R0.1-SNAPSHOT.jar:?]
	at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2784) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.network.protocol.game.PacketPlayInWindowClick.a(SourceFile:71) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.network.protocol.game.PacketPlayInWindowClick.a(SourceFile:14) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$ensureRunningOnSameThread$0(PlayerConnectionUtils.java:35) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.server.TickTask.run(SourceFile:18) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.util.thread.IAsyncTaskHandler.d(SourceFile:164) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(SourceFile:23) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1319) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:210) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.util.thread.IAsyncTaskHandler.B(SourceFile:138) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.server.MinecraftServer.bv(MinecraftServer.java:1298) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.server.MinecraftServer.B(MinecraftServer.java:1291) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.util.thread.IAsyncTaskHandler.b(SourceFile:147) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1248) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.server.MinecraftServer.x_(MinecraftServer.java:1258) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.server.MinecraftServer.y(MinecraftServer.java:1101) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:329) ~[spigot-1.21.3-R0.1-SNAPSHOT.jar:4383-Spigot-e65d67a-dd08a71]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
```

The issue is related to the sound effects inside the config no longer being loaded and played correctly after some change in Spigot. The next plugin uppdate should resolve this. You can try out this snapshot and let me know if this resolve the issue: https://github.com/Shopkeepers/Shopkeepers/actions/runs/11926967737/artifacts/2211272065

**Comment by blablubbabc** [comment]
This should be resolved in Shopkeepers v2.23.2
